### PR TITLE
[GS-117] Reduce number of requests made to Sheets API

### PR DESCRIPTION
**Pull request by naveen-nathan**
### Jira Ticket

[Jira Ticket](https://dashbd.atlassian.net/browse/<ticket-id>)

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] My branch name matches the format: `<ticket-id>/<brief-description-of-change>`
- [x] My PR name matches the format: `[<ticket-id>] <brief-description-of-change>`
- [ ] I have added doc-comments to all new functions ([JSDoc](https://jsdoc.app/) for JS and [Docstrings](https://peps.python.org/pep-0257/) for Python)
- [x] I have reviewed all of my code
- [x] My code only contains major changes related to my ticket

### CODE REVIEW

File: gradescopeCronJob/gradescope_to_spreadsheet.py [hunk 1]
```diff
@@ -98,10 +101,11 @@ def get_sub_sheet_titles_to_ids(sheet_api_instance):
     global subsheet_titles_to_ids
     if subsheet_titles_to_ids:
         return subsheet_titles_to_ids
+    print("Retrieving subsheet titles to ids")
     sheets = sheet_api_instance.get(spreadsheetId=SPREADSHEET_ID, fields='sheets/properties').execute()
-    sub_sheet_titles_to_ids = {sheet['properties']['title']: sheet['properties']['sheetId'] for sheet in
```
Comment (naveen-nathan) [inline gradescopeCronJob/gradescope_to_spreadsheet.py:110]: This PR is to rectify this minor mistake: I accidentally created a new local variable when I meant to use a similarly named global  variable.

Comment (Mehul-Gandhi) [reply gradescopeCronJob/gradescope_to_spreadsheet.py:110]: Have you tested this?

Comment (Mehul-Gandhi) [reply gradescopeCronJob/gradescope_to_spreadsheet.py:110]: Can you also do me a favor and get rid of `.gitignore` in the root folder, which has `oldScripts/.env` in it. We deleted the oldScripts folder.

File: gradescopeCronJob/gradescope_to_spreadsheet.py [hunk 2]
```diff
@@ -82,7 +85,7 @@ def writeToSheet(sheet_api_instance, assignment_scores, assignment_name = ASSIGN
             sheet_id = response['replies'][0]['addSheet']['properties']['sheetId']
         else:
             sheet_id = sub_sheet_titles_to_ids[assignment_name]
-        time.sleep(5)
+        time.sleep(SLEEP_TIME)
         update_sheet_with_csv(assignment_scores, sheet_api_instance, sheet_id)
         print("Successfully updated spreadsheet with new score data")
```
Comment (Mehul-Gandhi) [inline gradescopeCronJob/gradescope_to_spreadsheet.py]: Can you change this print statement to be more specific? For example, what assignment just got updated?

Comment (Mehul-Gandhi) [reply gradescopeCronJob/gradescope_to_spreadsheet.py]: This has been resolved by Manan's code. Please fetch the latest changes from main.

**Review by Mehul-Gandhi — CHANGES_REQUESTED** [review]
1. Pull from main
2. Retest your changes
3. Can you also do me a favor and get rid of .gitignore in the root folder, which has oldScripts/.env in it. We deleted the oldScripts folder

File: gradescopeCronJob/gradescope_to_spreadsheet.py (hunk 1 above)
Comment (naveen-nathan) [reply gradescopeCronJob/gradescope_to_spreadsheet.py:110]: I have tested it! The script now runs in less than 3 minutes.

Comment (naveen-nathan) [reply gradescopeCronJob/gradescope_to_spreadsheet.py:110]: Will remove the .gitignore

File: gradescopeCronJob/gradescope_to_spreadsheet.py (hunk 2 above)
Comment (naveen-nathan) [reply gradescopeCronJob/gradescope_to_spreadsheet.py]: Great! Thank you, @mananb77 ! Will do.

**Comment by naveen-nathan** [comment]
> 1. Pull from main
> 2. Retest your changes
> 3. Can you also do me a favor and get rid of .gitignore in the root folder, which has oldScripts/.env in it. We deleted the oldScripts folder

Will do!

**Comment by naveen-nathan** [comment]
> 1. Pull from main
> 2. Retest your changes
> 3. Can you also do me a favor and get rid of .gitignore in the root folder, which has oldScripts/.env in it. We deleted the oldScripts folder

Done! Retested after pulling from main. It works! Thanks for the review, @Mehul-Gandhi! Thanks for adding a config file, @mananb77 !

**Review by Mehul-Gandhi — APPROVED** [review]
LGTM

**Comment by naveen-nathan** [comment]
> Thanks

Thanks, Mehul! Will merge now!

**Pull request merged**